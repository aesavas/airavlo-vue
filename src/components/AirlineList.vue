<script setup lang="ts">
defineProps({
  majorAirlines: Array<string>
})
</script>

<template>
  <div
    class="accordion mt-2 mb-3"
    id="accordion"
  >
    <div class="accordion-item">
      <h2 class="accordion-header">
        <button
          class="accordion-button collapsed"
          type="button"
          data-bs-toggle="collapse"
          data-bs-target="#flush-collapseOne"
          aria-expanded="false"
          aria-controls="flush-collapseOne"
        >
          <p><strong>Major Airlines:</strong></p>
        </button>
      </h2>
      <div
        id="flush-collapseOne"
        class="accordion-collapse collapse"
        data-bs-parent="#accordion"
      >
        <div class="accordion-body">
          <div
            v-for="value in majorAirlines"
            :key="value"
          >ðŸ›« {{ value }}</div>
        </div>
      </div>
    </div>
  </div>
</template>
