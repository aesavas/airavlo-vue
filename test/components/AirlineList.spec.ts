import AirlineList from "@/components/AirlineList.vue";
import { mount } from "@vue/test-utils";
import { describe, expect, it } from "vitest";

describe("Airline List Component", () => {
  it("render airline list component", () => {
    const wrapper = mount(AirlineList, {
      props: {
        majorAirlines: ["Foo Airlines", "Bar Airlines"]
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
