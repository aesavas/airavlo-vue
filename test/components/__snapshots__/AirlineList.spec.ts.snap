// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Airline List Component > render airline list component 1`] = `
"<div class="accordion mt-2 mb-3" id="accordion">
  <div class="accordion-item">
    <h2 class="accordion-header"><button class="accordion-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseOne" aria-expanded="false" aria-controls="flush-collapseOne">
        <p><strong>Major Airlines:</strong></p>
      </button></h2>
    <div id="flush-collapseOne" class="accordion-collapse collapse" data-bs-parent="#accordion">
      <div class="accordion-body">
        <div>ğŸ›« Foo Airlines</div>
        <div>ğŸ›« Bar Airlines</div>
      </div>
    </div>
  </div>
</div>"
`;
